### DISCUSSION TITLE
Integration into a Qt/QML application

### DISCUSSION THREAD

**Discussion by mcallegari** [o3de/o3de#7234] [Q&A]
Hello there!
I'm quite impressed by this project and I'm wondering if there is any chance for me to try O3DE into a Qt/QML application.
Right now I'm using the Qt3D module but performances suck and it's kind of in an abandoned state.
I need to make real time lighting simulation, like this (or better than this!):
https://youtu.be/eI_NfA_vyA0
https://youtu.be/yoQVzYR-NwM

Thanks in advance!

**Comment by mcallegari** [comment]
More than a year later...no interest or insight on this? Thanks

**Reply by cgalvan** [reply]
Hello! It is likely possible, although there would be an unknown amount of dependencies in order to be able to drop an atom viewport widget into a completely different application. A good starting point would be looking at the `EditorViewportWidget` class (located in `Code\Editor\EditorViewportWidget.cpp`), which is the top-level `QWidget` that embeds the atom viewport into the `Editor` application.

Any thoughts from @o3de/sig-graphics-audio ?
